### PR TITLE
all needed controllers, dtos and mappers for them were implemented

### PR DESCRIPTION
**Pull request by oleksiistepaniak**

### CODE REVIEW

File: src/main/java/mate/academy/spring/model/dto/request/OrderRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package mate.academy.spring.model.dto.request;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class OrderRequestDto {
```
Comment (potrap) [inline src/main/java/mate/academy/spring/model/dto/request/OrderRequestDto.java:6]: Do you need this class?

**Review by potrap — APPROVED** [review]
gj!

**Review by NickolayVoloshchuk — APPROVED** [review]
GJ)

File: src/main/java/mate/academy/spring/controller/AuthenticationController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final DtoResponseMapper<UserResponseDto, User> userResponseMapper;
+
+    public AuthenticationController(
+            AuthenticationService authenticationService,
+            UserResponseMapper userResponseMapper
+    ) {
```
Comment (Andriy-dz) [inline src/main/java/mate/academy/spring/controller/AuthenticationController.java:21]:
```suggestion
            UserResponseMapper userResponseMapper) {

```

File: src/main/java/mate/academy/spring/controller/AuthenticationController.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        this.authenticationService = authenticationService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @PostMapping("/register")
+    public UserResponseDto register(@RequestBody UserRequestDto userRequestDto) {
+        return userResponseMapper.toDto(authenticationService
```
Comment (Andriy-dz) [inline src/main/java/mate/academy/spring/controller/AuthenticationController.java:28]: add a check whether such a user does not exist

File: src/main/java/mate/academy/spring/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+        this.shoppingCartResponseMapper = shoppingCartResponseMapper;
+    }
+
+    @PutMapping("/movie-sessions")
+    public ShoppingCartResponseDto addMovieSession(
+            @RequestParam Long userId,
+            @RequestParam Long movieSessionId
+    ) {
```
Comment (Andriy-dz) [inline src/main/java/mate/academy/spring/controller/ShoppingCartController.java:40]:
```suggestion
            @RequestParam Long movieSessionId) {

```

File: src/main/java/mate/academy/spring/mapper/impl/response/ShoppingCartResponseMapper.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.Ticket;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShoppingCartResponseMapper implements
+        DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> {
+
```
Comment (Andriy-dz) [inline src/main/java/mate/academy/spring/mapper/impl/response/ShoppingCartResponseMapper.java:13]: empty line

File: src/main/java/mate/academy/spring/model/dto/request/OrderRequestDto.java (hunk 1 above)
Comment (Andriy-dz) [reply src/main/java/mate/academy/spring/model/dto/request/OrderRequestDto.java:6]: agree with @potrap

**Review by Andriy-dz — CHANGES_REQUESTED** [review]
Good Job! Let`s improve your solution!

File: src/main/java/mate/academy/spring/controller/AuthenticationController.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    public AuthenticationController(
+            AuthenticationService authenticationService,
+            UserService userService, UserResponseMapper userResponseMapper) {
+        this.authenticationService = authenticationService;
+        this.userService = userService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @PostMapping("/register")
+    public UserResponseDto register(@RequestBody UserRequestDto userRequestDto) {
```
Comment (nacenik) [inline src/main/java/mate/academy/spring/controller/AuthenticationController.java:30]: This method should return `void` 
image

**Review by nacenik — APPROVED** [review]
Great job! You can improve your solution if you want 😉

**Pull request closed without merging**